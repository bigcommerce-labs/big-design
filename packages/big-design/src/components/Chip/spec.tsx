import { fireEvent, render } from '@testing-library/react';
import 'jest-styled-components';
import React from 'react';

import { Chip } from './index';

test('renders correctly without the display icon', () => {
  const { container } = render(<Chip displayCloseIcon={false} id={1} label="Test Label" />);

  expect(container.firstChild).toMatchSnapshot();
});

test('renders correctly with the display icon', () => {
  const { container } = render(<Chip displayCloseIcon={true} onRemove={jest.fn()} id={1} label="Test Label" />);

  expect(container.firstChild).toMatchSnapshot();
});

test('calls the remove method if clicked', () => {
  const onRemove = jest.fn();
  const { container } = render(<Chip displayCloseIcon={true} onRemove={onRemove} id={1} label="Test Label" />);

  const icon = container.getElementsByTagName('svg');

  fireEvent.click(icon[0]);
  expect(onRemove).toHaveBeenCalled();
});
