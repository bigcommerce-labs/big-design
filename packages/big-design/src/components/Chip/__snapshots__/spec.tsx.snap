// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with the display icon 1`] = `
.c1 {
  vertical-align: middle;
  height: 1rem;
  width: 1rem;
}

.c0 {
  background-color: #D9DCE9;
  border-radius: 2px;
  color: #B4BAD1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  line-height: 1.5rem;
  margin: 0.25rem;
  padding: 0 0.5rem;
  z-index: 1;
}

.c0 svg {
  cursor: pointer;
  left: 0.25rem;
  position: relative;
  top: 0.25rem;
}

<div
  class="c0"
>
  Test Label
  <svg
    class="c1"
    fill="currentColor"
    height="24"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="24"
  >
    <title />
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M18.3 5.71a.996.996 0 00-1.41 0L12 10.59 7.11 5.7A.996.996 0 105.7 7.11L10.59 12 5.7 16.89a.996.996 0 101.41 1.41L12 13.41l4.89 4.89a.996.996 0 101.41-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
    />
  </svg>
</div>
`;

exports[`renders correctly without the display icon 1`] = `
.c0 {
  background-color: #D9DCE9;
  border-radius: 2px;
  color: #B4BAD1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  line-height: 1.5rem;
  margin: 0.25rem;
  padding: 0 0.5rem;
  z-index: 1;
}

.c0 svg {
  cursor: pointer;
  left: 0.25rem;
  position: relative;
  top: 0.25rem;
}

<div
  class="c0"
>
  Test Label
</div>
`;
